### PR TITLE
use automethod for LightningModule method

### PR DESCRIPTION
**Pull request by ydcjeff**
## What does this PR do?

Replace `autofunction` with `automethod` in LightningModule API part of LightningModule page.

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

@Borda

### CODE REVIEW

**Review by Borda — APPROVED** [review]
lgtm

**Review by SkafteNicki — APPROVED** [review]
LGTM

**Review by SeanNaren — APPROVED** [review]
Looks good!

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -399,7 +399,7 @@ def training_step(self, *args, **kwargs):
                 :paramref:`~pytorch_lightning.trainer.trainer.Trainer.truncated_bptt_steps` > 0.
 
         Return:
-            roch.Tensor or a dictionary with anything you want (must include the keyword 'loss')
+            :class:`~torch.Tensor` or a dictionary with anything you want (must include the keyword 'loss')
```
Comment (rohitgr7) [inline pytorch_lightning/core/lightning.py:402]:
```suggestion
            Either a :class:`~torch.Tensor` or a dictionary with anything
                you want (must include the keyword 'loss') or None
```
I believe now you can skip training_step by returning None.

**Comment by williamFalcon** [comment]
This doesn't look great. 

image

Please keep autofunctions.

**Comment by ydcjeff** [comment]
@williamFalcon you need to reinstall the lightning sphinx theme, so it will appear like autofunction

**Comment by awaelchli** [comment]
@ydcjeff that makes sense, explains why in CI build it looked fine :) 
@williamFalcon the benefit of using automethod is that you will see this:

`LightningModule.training_step(*args, **kwargs)`

instead of currently 

`pytorch_lightning.core.lightning.LightningModule.training_step(*args, **kwargs)`

**Comment by williamFalcon** [comment]
but it was also like that on master. here, let’s merge and compare. 

if it doesn’t work please revert

**Comment by awaelchli** [comment]
https://pytorch-lightning.readthedocs.io/en/latest/lightning_module.html#training-step
Looking good on latest read the docs 🎉

**Pull request merged**